pkg/operator/operator.go

```diff
@@ -475,9 +475,18 @@ func (o *Operator) sync(key string) error {
 		klog.Errorf("error occurred while setting status to in progress: %v", err)
 	}
 
-	taskName, err := tl.RunAll()
-	if err != nil {
-		o.reportError(err, taskName)
+	taskErrors := tl.RunAll()
+	if len(taskErrors) > 0 {
+		err := errors.New("Cluster monitoring update failed.")
+		var failedTask string
+		if len(taskErrors) == 1 {
+			failedTask = strings.Join(strings.Fields(taskErrors[0].Name+"Failed"), "")
+		} else {
+			failedTask = "MultipleTasksFailed"
+		}
+
+		err = errors.Wrap(err, failedTask)
+		o.reportError(taskErrors, failedTask)
 		return err
```

Comment:
my previous comment was to imply that we could squeeze the code and avoid the intermediate `err` declaration. Sorry it wasn't clear :)

```suggestion
		return errors.Formatf("cluster monitoring update failed: %s", failedTask)
```